src/Storages/ObjectStorageQueue/StorageObjectStorageQueue.cpp

```diff
@@ -204,7 +204,8 @@ StorageObjectStorageQueue::StorageObjectStorageQueue(
     storage_metadata.setColumns(columns);
     storage_metadata.setConstraints(constraints_);
     storage_metadata.setComment(comment);
-    storage_metadata.settings_changes = engine_args->settings->ptr();
+    if (engine_args->settings)
```

Comment:
This is a fix for crash from PR https://github.com/ClickHouse/ClickHouse/pull/73274, will update the branch once that PR is merged so it will no longer be in diff here.